source/contribute.rst

```diff
@@ -267,76 +347,67 @@ sense to revert a commit documenting a feature without also reverting
-**maintainer** to review and pull into **doctrine** as appropriate:
-
-.. code-block:: console
-
-    $ git push origin fix-weird-bug
-    To git@github.com:hobodave/dbal.git
-        * [new branch]      fix-weird-bug -> fix-weird-bug
-
-Now you are ready to send a pull request from this branch and ask for a
-review from a maintainer.
+Basically, a rebase takes all changes on your topic branch and moves
+them to another starting point. This starting point was the Doctrine
+branch that you chose when you created your topic branch, at that point
+in time. The rebase will move your changes to be based on the current
+state of this branch.
 
-Topic Branch Cleanup
---------------------
-
-Once your work has been merged by the branch maintainer, it will no
-longer be necessary to keep the local branch or remote branch, so you
-can remove them!
+Here is how to proceed if you need to rebase on ``2.3.x``:
 
-Sync your local 2.11.x branch:
+1. Fetch all new commits: `git fetch doctrine``.
+2. Rebase on what you fetched:
+   ``git rebase doctrine/2.3.x``
+3. If you run into a conflict, fix it with ``git mergetool``, then
+   continue on your merry way with ``git rebase --continue``.
+4. Force push to overwrite the previous version : ``git push --force``.
 
-.. code-block:: console
+.. _rebasing-on-another-branch:
 
-    $ git checkout 2.11.x
-    $ git pull --rebase
+On another branch
+~~~~~~~~~~~~~~~~~
 
-Remove your local topic branch using -d to ensure that it has been merged by
-upstream. Branch -d will not delete a branch that is not an ancestor of
-your current head.
+Another case where you need a rebase is when you want to change the
+target branch of your PR. For instance, you might have created your PR
+against ``2.3.x`` but you are told to change it to ``2.4.x``. In that
+case, if ``2.4.x`` does not have all the commits that ``2.3.x`` has, you
+need to drop these commits while rebasing: it is not your responsibility
+to merge them up.
 
-From the git-branch man page:
+To see if you need to get rid of any commits, you can click on the
+``Edit`` button next to the PR title, and change the target branch to
+``2.4.x``. If you see commits that are not yours, you need to rebase.
 
-.. code-block:: console
+Here is how to proceed if you need to rebase on ``2.4.x`` and drop the
+commits you did not author:
 
-    -d
-        Delete a branch. The branch must be fully merged in HEAD.
-    -D
-        Delete a branch irrespective of its merged status.
+1. Fetch all new commits: `git fetch doctrine``.
+2. Rebase on what you fetched:
+   ``git rebase --interactive doctrine/2.3.4 # use -i for short``
+3. You will be presented with a list of commits. You need to drop all
+   commits that are not yours. You can do that by replacing ``pick`` with
+   ``drop``.
+4. If you run into a conflict, fix it with ``git mergetool``, then
+   continue on your merry way with ``git rebase --continue``.
+5. Force push to overwrite the previous version : ``git push --force``.
 
```

Comment:
6. Update the Pull Request target in GitHub

Reply:
Already mentioned above (and I think it's best to do it before pushing)